totalRP3/Core/Color.lua

```diff
@@ -418,33 +377,17 @@ function ConvertHWBToRGB(h, w, b)
 end
 
 function ConvertRGBToHSL(r, g, b)
-	local cmax = math.max(r, g, b);
-	local cmin = math.min(r, g, b);
-	local c = cmax - cmin;
-
-	local h = 0;
-	local s = 0;
-	local l = (cmin + cmax) / 2;
-
-	if c ~= 0 then
-		s = (l == 0 or l == 1) and 0 or ((cmax - l) / math.min(l, 1 - l));
+	return C_ColorUtil.ConvertHSVToHSL(ConvertRGBToHSV(r, g, b));
+end
 
-		if cmax == r then
-			h = (g - b) / c + (g < b and 6 or 0);
-		elseif cmax == g then
-			h = (b - r) / c + 2;
-		else
-			h = (r - g) / c + 4;
-		end
+function ConvertRGBToHSV(r, g, b)
+	local h, s, v = C_ColorUtil.ConvertRGBToHSV(r, g, b);
 
-		h = h / 6;
+	if h == -1 then
+		h = 0;  -- Achromatic input; clamp to zero hue.
```

Comment:
RGB to HSV conversion has a documented note that it'll return a -1 value for hue for pure white or black inputs. Our old code here I believe clamped such cases to always use a hue of zero, so retain this helper function that applies that conversion.